### DISCUSSION TITLE
Optimize build process

### DISCUSSION THREAD

**Discussion by apriljunge** [railtoolkit/tikz-trackschematic#17] [Ideas]
I just tried to set up a development environment and have a few suggestions on how to improve the process.
Let's discuss if they work and what you think about them. Maybe it would make sense to split the tasks if you agree that it should be done.

1. have one unified build script
This build script could have a parameter like `--target` where you can specify if you want to build something specific (docs, snippets, ...) or all. If not specified the script could ask what to do.
2. move all scripts into root
You don't have to cd for tests etc. Most build scripts could be unified into one (see 1.)
3. Replace copying with environment variables
I am really unsure about this, because I do not know what's exaclty the magic behind copying the files to different folders. It didn't work for me. I could fix it by setting the environment vairable `TEXINPUTS` (example below for test.sh). Maybe this snippet could replace the copy process? Then maybe the script doesn't need super user rights any more? The environment variable modifies the directories where pdflatex looks for sty files ([stackexchange discussion](https://tex.stackexchange.com/questions/93712/definition-of-the-texinputs-variable))
```
export TEXINPUTS=.:../src/:$TEXINPUTS
```
4. Enhance script usage description
The information scripts give about their parameter didn't help me very much: "Usage: dev-install.sh [-q] [-b]".
Maybe this could be enhanced a little.
5. Enhance developement documentation
Give short introduction in commands, what you need installed etc. Could be in CONTRIBUTING.md oder README.md

**Comment by kaat0** [comment]
1. interesting idea. I will have a look into it!
2. obvious after 1.
3. the currenct approach results from my need to have a TeX Live distrubtion version besides the development version. The package option [dev] switches between the version. For this there needs to be two differently named tikz-trackschematic.sty files.
4. created new issue #20
5. created new issue #21

**Reply by apriljunge** [reply]
Do you need both versions in the same project? Or the dev one for this repo and the distribution version for other projects?

**Reply by kaat0** [reply]
I use a development version for enhancing the package and a the TeX Live version for production so that errors I make in the development do not effect my other works.
